### ISSUE TITLE
CapabilityBasedPartitioner doesn't merge subgraphs for several outputs

### ISSUE THREAD

**Issue by IvanYashchuk** [pytorch/pytorch#86108]
### 🐛 Describe the bug

In the PR description that added `CapabilityBasedPartitioner` (https://github.com/pytorch/pytorch/pull/79439) it's said that
>  this partitioner tries to form the largest subgraphs that only contain the supported ops.

Here's an example where the largest subgraph is not formed

```py
import torch
from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner
from torch.fx.experimental.proxy_tensor import make_fx
from torch._prims.context import TorchRefsMode
from copy import deepcopy

class PrimsOperatorSupport(torch.fx.passes.operator_support.OperatorSupport):
    def is_node_supported(self, submodules, node: torch.fx.Node) -> bool:
        return (
            node.op == "call_function"
            and str(node.target).startswith("prims.")
        )

def func(input):
    return torch.sin(input**2), torch.cos(input**2)

a = torch.randn(2, 4, 32, 16, dtype=torch.float32, device="cuda")

with TorchRefsMode():
    gm = make_fx(func)(a)

partitioned_gm = deepcopy(gm)
supported_ops = PrimsOperatorSupport()
partitioner = CapabilityBasedPartitioner(
    partitioned_gm, supported_ops
)
partitions = partitioner.propose_partitions()
partitioner.fuse_partitions(partitions)

print(f"len(partitions) = {len(partitions)}")
print(partitions)

```
The length of `partitions` is 2, but should be 1, because all functions in the graph are supported and should be merged into one subgraph.
```py
# gm.graph.print_tabular()
opcode         name     target             args                kwargs
-------------  -------  -----------------  ------------------  --------
placeholder    input_1  input_1            ()                  {}
call_function  mul      prims.mul.default  (input_1, input_1)  {}
call_function  sin      prims.sin.default  (mul,)              {}
call_function  mul_1    prims.mul.default  (input_1, input_1)  {}
call_function  cos      prims.cos.default  (mul_1,)            {}
output         output   output             ((sin, cos),)       {}
```
```py
# partitioned_gm.graph.print_tabular()
opcode       name     target    args                   kwargs
-----------  -------  --------  ---------------------  --------
placeholder  input_1  input_1   ()                     {}
call_module  fused_0  fused_0   (input_1,)             {}
call_module  fused_1  fused_1   (input_1,)             {}
output       output   output    ((fused_1, fused_0),)  {}
```

### Versions

Latest master.

cc @ezyang @SherlockNoMad @soumith

**Comment by IvanYashchuk** [comment]
Thank you! Marking "input" as supported does fix the problem for this example.

Is the "output" node need to be marked as supported in general? What are the practical situations where "placeholder" nodes need to be marked as unsupported?

**Comment by jjsjann123** [comment]
> Actually no, marking "placeholder" nodes as supported is incorrect. CapabilityBasedPartitioner generates an invalid graph with no placeholders then (hence taking 0 arguments):

I don't think we can take `placeholder`, which marks the input for the graph and taking that away would change the graph.
Another thing is, we would prefer to have sibling fusion instead of relying on always have the producer fusible.

**Comment by jjsjann123** [comment]
This should be patched in #86511
